cpp/src/ray/config_internal.cc

```diff
@@ -60,6 +60,15 @@ void ConfigInternal::Init(RayConfig &config, int *argc, char ***argv) {
   if (config.redis_password_) {
     redis_password = *config.redis_password_;
   }
+  if (config.num_cpus > 0) {
+    num_cpus = config.num_cpus;
+  }
+  if (config.num_gpus > 0) {
+    num_gpus = config.num_gpus;
```

Comment:
What if a user wants to start a node with 0 CPUs or 0 GPUs? In Python we use `None` for default value, not 0.